backend/src/services/complectations/complectations.service.ts

```diff
@@ -0,0 +1,85 @@
+import { prisma } from '@data/prisma-client';
+
+import type {
+  ModelComplectationsResponseDto,
+  ComplectationsResponseDto,
+  OptionType,
+} from '@autoline/shared/common/types/types';
+
+const getComplectationsById = async (
+  complectationId: string | undefined,
+): Promise<ComplectationsResponseDto> => {
+  const data = (await prisma.complectation.findUnique({
+    where: {
+      id: complectationId,
+    },
+    select: {
+      prices_ranges: {
+        select: {
+          price_start: true,
+          price_end: true,
+        },
+      },
+      color: { select: { name: true } },
+      engine_displacement: true,
+      engine_power: true,
+      drivetrain: { select: { name: true } },
+      fuel_type: { select: { name: true } },
+      transmission_type: { select: { name: true } },
+      options: {
+        select: {
+          option: { select: { name: true, type: true } },
+          important: true,
+        },
+      },
+    },
+  })) as ModelComplectationsResponseDto;
+
+  const optionType = [
+    'security',
+    'optics',
+    'multimedia',
+    'upholstery',
+    'sound',
+    'design',
+    'comfort',
+    'auxiliary',
+  ];
+
+  const options: OptionType = {
+    security: [],
+    optics: [],
+    multimedia: [],
+    upholstery: [],
+    sound: [],
+    design: [],
+    comfort: [],
+    auxiliary: [],
+    important: [],
+  };
+
+  optionType.forEach((optionName) => {
+    data?.options.forEach((option) => {
```

Comment:
I prefer to use reduce here
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce